lib/mongodb.go

```diff
@@ -322,22 +181,37 @@ func (m MongoDBPlugin) LabelPrefix() string {
 
 // Do the plugin
 func Do() {
-	optHost := flag.String("host", "localhost", "Hostname")
-	optPort := flag.String("port", "27017", "Port")
+	optHost := flag.String("host", "", "Hostname")
+	optPort := flag.String("port", "", "Port")
 	optUser := flag.String("username", "", "Username")
 	optPass := flag.String("password", os.Getenv("MONGODB_PASSWORD"), "Password")
 	optSource := flag.String("source", "", "authenticationDatabase")
+	optURL := flag.String("url", os.Getenv("MONGODB_URL"), "URL (exclusive)")
 	optVerbose := flag.Bool("v", false, "Verbose mode")
 	optTempfile := flag.String("tempfile", "", "Temp file name")
 	optKeyPrefix := flag.String("metric-key-prefix", "", "Metric key prefix")
 	flag.Parse()
 
 	var mongodb MongoDBPlugin
 	mongodb.Verbose = *optVerbose
-	mongodb.URL = net.JoinHostPort(*optHost, *optPort)
+
+	mongodb.RawURL = *optURL
+
+	if *optURL != "" && (*optHost != "" || *optPort != "" || *optUser != "" || *optPass != "" || *optSource != "") {
+		fmt.Println("-url is exclusive.")
```

Comment:
[may] Same as the point about the usage of URL. How about the following expression?

> -url option is mutually exclusive with these options: -host, -port, -username, -password, and -source

Reply:
This configuration can be set using the environment variable. However, this message only considers command line options.

Reply:
applied your suggestion message.